pkg/cloud/services/eks/cluster.go

```diff
@@ -523,22 +523,23 @@ func (s *Service) reconcileEKSEncryptionConfig(currentClusterConfig []*eks.Encry
 	encryptionConfigs := s.scope.ControlPlane.Spec.EncryptionConfig
 	updatedEncryptionConfigs := makeEksEncryptionConfigs(encryptionConfigs)
 
-	switch {
-	case compareEncryptionConfig(currentClusterConfig, updatedEncryptionConfigs):
+	if compareEncryptionConfig(currentClusterConfig, updatedEncryptionConfigs) {
 		s.V(2).Info("encryption configuration unchanged, no action")
 		return nil
-	case len(currentClusterConfig) == 0 && len(updatedEncryptionConfigs) > 0:
+	}
+
+	if len(currentClusterConfig) == 0 && len(updatedEncryptionConfigs) > 0 {
 		s.V(2).Info("enabling encryption for eks cluster", "cluster", s.scope.KubernetesClusterName())
 		if err := s.updateEncryptionConfig(updatedEncryptionConfigs); err != nil {
 			record.Warnf(s.scope.ControlPlane, "FailedUpdateEKSControlPlane", "failed to update the EKS control plane encryption configuration: %v", err)
 			return errors.Wrapf(err, "failed to update EKS cluster")
 		}
-	default:
-		record.Warnf(s.scope.ControlPlane, "FailedUpdateEKSControlPlane", "failed to update the EKS control plane: disabling EKS encryption is not allowed after it has been enabled")
-		return errors.Errorf("failed to update the EKS control plane: disabling EKS encryption is not allowed after it has been enabled")
+
+		return nil
 	}
 
-	return nil
+	record.Warnf(s.scope.ControlPlane, "FailedUpdateEKSControlPlane", "failed to update the EKS control plane: disabling EKS encryption is not allowed after it has been enabled")
+	return errors.Errorf("failed to update the EKS control plane: disabling EKS encryption is not allowed after it has been enabled")
```

Comment:
Once set, modifying or deleting the config is not allowed in the webhook, so not really needed.

Reply:
@richardcase What was the fix here? I could not see.

Reply:
Agreed it should never get here....although it was with the issue that this PR relates to. I'd be tempted to keep it as it would be good to get an error .